src/components/numeral-date/numeral-date.test.tsx

```diff
@@ -0,0 +1,1995 @@
+    expect(
+      screen.getByText("Day should be a number within a 1-31 range.")
+    ).toBeVisible();
+  });
+
+  it("should render the validation message when the user types a value in the `Day` field less than 1 and tabs", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "", mm: "", yyyy: "" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(dayInput, "0");
+    await user.tab();
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+    expect(
+      screen.getByText("Day should be a number within a 1-31 range.")
+    ).toBeVisible();
+  });
+
+  it("should render the validation message when the user types `02` in the `Month` field and tabs when the `Day` field has a value greater than 28", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "29", mm: "", yyyy: "2021" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    const dayInput = screen.getByRole("textbox", { name: "Day" });
+    const yearInput = screen.getByRole("textbox", { name: "Year" });
+    await user.type(monthInput, "02");
+    await user.tab();
+
+    expect(dayInput).toBeInvalid();
+    expect(monthInput).toBeInvalid();
+    expect(yearInput).toBeInvalid();
+    expect(
+      screen.getByText("Day in February should be a number within 1-28.")
+    ).toBeVisible();
+  });
+
+  it("should not render the validation message when the user types '02' in the`Month` field and tabs when the `Day` field has a valid value", async () => {
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+    render(
+      <CarbonProvider validationRedesignOptIn>
+        <NumeralDate
+          value={{ dd: "29", mm: "", yyyy: "2020" }}
+          onChange={() => {}}
+          enableInternalError
+        />
+      </CarbonProvider>
+    );
+
+    const monthInput = screen.getByRole("textbox", { name: "Month" });
+    await user.type(monthInput, "02");
+    await user.tab();
+
+    expect(
+      screen.queryByText("Day in February should be a number within 1-28.")
+    ).not.toBeInTheDocument();
+  });
+
+  it("should use the current year and not render the validation message when the user types '02' in the`Month` field and tabs when the `Day` field has a valid value", async () => {
```

Comment:
nittiest of nitpicks:

```suggestion
  it("should use the current year and not render the validation message when the user types '02' in the `Month` field and tabs when the `Day` field has a valid value", async () => {
```